tests/e2e/vfile_demote_promote_test.go

```diff
@@ -0,0 +1,183 @@
+
+	log.Printf("Wait 20 seconds for new manager to be updated")
+	time.Sleep(20 * time.Second)
+
+	err = dockercli.DemoteNode(s.worker1, s.master)
+	c.Assert(err, IsNil, Commentf("Failed to demote manager %s", s.master))
+
+	log.Printf("Wait 20 seconds for new worker to be updated")
+	time.Sleep(20 * time.Second)
+
+	out, err = dockercli.CreateVFileVolume(s.master, s.volName2)
+	c.Assert(err, IsNil, Commentf(out))
+
+	accessible = verification.CheckVolumeAvailability(s.worker1, s.volName2)
+	c.Assert(accessible, Equals, true, Commentf("Volume %s is not available", s.volName1))
+
+	out, err = dockercli.AttachVFileVolume(s.master, s.volName2, s.container2Name)
+	c.Assert(err, IsNil, Commentf(out))
+
+	out = verification.GetVFileVolumeGlobalRefcount(s.volName1, s.master)
+	grefc, _ := strconv.Atoi(out)
+	c.Assert(grefc, Equals, 1, Commentf("Expected volume %s global refcount to be 1, found %s", s.volName1, out))
+
+	out = verification.GetVFileVolumeGlobalRefcount(s.volName2, s.worker1)
+	grefc, _ = strconv.Atoi(out)
+	c.Assert(grefc, Equals, 1, Commentf("Expected volume %s global refcount to be 1, found %s", s.volName2, out))
+
+	out, err = dockercli.RemoveContainer(s.worker1, s.container1Name)
+	c.Assert(err, IsNil, Commentf(out))
+
+	out = verification.GetVFileVolumeGlobalRefcount(s.volName1, s.master)
+	grefc, _ = strconv.Atoi(out)
+	c.Assert(grefc, Equals, 0, Commentf("Expected volume %s global refcount to be 0, found %s", s.volName1, out))
+
+	out, err = dockercli.RemoveContainer(s.master, s.container2Name)
+	c.Assert(err, IsNil, Commentf(out))
+
+	out = verification.GetVFileVolumeGlobalRefcount(s.volName2, s.worker1)
+	grefc, _ = strconv.Atoi(out)
+	c.Assert(grefc, Equals, 0, Commentf("Expected volume %s global refcount to be 0, found %s", s.volName2, out))
+
+	log.Printf("Wait 20 seconds for volume status back to Ready")
+	time.Sleep(20 * time.Second)
+
+	out = verification.GetVFileVolumeStatusHost(s.volName1, s.master)
+	log.Println("GetVFileVolumeStatusHost return out[%s] for volume %s", out, s.volName1)
+	c.Assert(out, Equals, "Ready", Commentf("Volume %s status is expected to be [Ready], actual status is [%s]",
+		s.volName1, out))
+
+	out = verification.GetVFileVolumeStatusHost(s.volName2, s.worker1)
+	log.Println("GetVFileVolumeStatusHost return out[%s] for volume %s", out, s.volName2)
+	c.Assert(out, Equals, "Ready", Commentf("Volume %s status is expected to be [Ready], actual status is [%s]",
+		s.volName2, out))
+
+	accessible = verification.CheckVolumeAvailability(s.master, s.volName1)
+	c.Assert(accessible, Equals, true, Commentf("Volume %s is not available", s.volName1))
+
+	accessible = verification.CheckVolumeAvailability(s.worker1, s.volName2)
+	c.Assert(accessible, Equals, true, Commentf("Volume %s is not available", s.volName2))
+
+	out, err = dockercli.DeleteVolume(s.master, s.volName1)
+	c.Assert(err, IsNil, Commentf(out))
+
+	out, err = dockercli.DeleteVolume(s.worker1, s.volName2)
+	c.Assert(err, IsNil, Commentf(out))
+
```

Comment:
I think code after this line is just to make reset the test bed, right? It is not part of the test itself. If it is true, please add a comment here.

Reply:
You can put it here, but just add a one line comment to say that the following code is for reset the testbed.